### PR TITLE
#TPROD-328 Migrate Stage Bundle to twig templates

### PR DESCRIPTION
**Pull request by SalemCode8**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #TPROD-328 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. in the left side menu click on "Stages"
3. Test every single functionality on this page. Creating/editing/cloning webhooks, filtering, pagination, etc. - all functionality should work exactly like it did before.

### CODE REVIEW

**Comment by SalemCode8** [comment]
Global Search views not tested as they blocked by #11566

File: app/bundles/StageBundle/Controller/StageController.php [hunk 1]
```diff
@@ -202,10 +202,10 @@ public function newAction($entity = null)
                 'viewParameters' => [
                     'tmpl'    => $this->request->isXmlHttpRequest() ? $this->request->get('tmpl', 'index') : 'index',
                     'entity'  => $entity,
-                    'form'    => $this->setFormTheme($form, 'MauticStageBundle:Stage:form.html.php', $themes),
+                    'form'    => $form->createView(), //Check if this is the correct way to do this: $this->setFormTheme($form, 'MauticStageBundle:Stage:form.html.twig', $themes),
```
Comment (escopecz) [inline app/bundles/StageBundle/Controller/StageController.php]: Is it the correct way?

**Review by escopecz — COMMENTED** [review]
Do we need the empty file `app/bundles/StageBundle/Views/Event/actions.html.twig`?

File: app/bundles/StageBundle/Controller/StageController.php (hunk 1 above)
Comment (SalemCode8) [reply app/bundles/StageBundle/Controller/StageController.php]: I think it's safe until the full conversion of the form themes

**Comment by SalemCode8** [comment]
> Do we need the empty file `app/bundles/StageBundle/Views/Event/actions.html.twig`?

I didn't know the use of the PHP one and search the whole application without a reference to it

File: app/bundles/StageBundle/Controller/StageController.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/StageBundle/Controller/StageController.php]: Does it change the UI or behavior anyhow?

**Comment by escopecz** [comment]
You are right. There is actions.html.php empty file as well. Let's remove it and test if it cause any problem.

File: app/bundles/StageBundle/Controller/StageController.php (hunk 1 above)
Comment (SalemCode8) [reply app/bundles/StageBundle/Controller/StageController.php]: No so far

Comment (escopecz) [reply app/bundles/StageBundle/Controller/StageController.php]: Ok. Let's remove this comment then and move forward with this :)

Comment (SalemCode8) [reply app/bundles/StageBundle/Controller/StageController.php]: Sure

**Review by escopecz — DISMISSED** [review]
I see no issue in the code. Thanks!

**Review by kuzmany — DISMISSED** [review]
I've made basic tests in UI. 
It works for me 👍

**Review by volha-pivavarchyk — DISMISSED** [review]
Works well

**Review by RCheesley — DISMISSED** [review]
Tested all functionality, working as expected! Thanks for the PR @SalemCode8 🚀

**Comment by RCheesley** [comment]
Not sure why we have the PHPSTAN failure - the file does not relate to this PR. Any thoughts @escopecz ?

**Comment by escopecz** [comment]
It does not seem to be related to this PR. I'll investigate

**Comment by escopecz** [comment]
The test is failing due to CodeCov networking issue even after 3rd rebuild. Merging.

**Pull request merged**